### ISSUE TITLE
Che addon: add support for multi user Che

### ISSUE THREAD

**Issue by l0rd** [minishift/minishift#2746]
### General information

Che minishift addon currently supports only [Che single user](https://www.eclipse.org/che/docs/openshift-single-user.html). We want to add support for [Che multi user](https://www.eclipse.org/che/docs/openshift-multi-user.html) too. 

This issue is part of the discussion we are having with Che community to drop Che single user support at all (c.f. https://github.com/eclipse/che/issues/10996). Anyway this issue is not about dropping the support for single user but adding an option to run Che multi user.

C.f. rh-che (multi user) minishift addon https://github.com/redhat-developer/rh-che/tree/master/openshift/minishift-addons

**Comment by LalatenduMohanty** [comment]
@l0rd Also with https://docs.okd.io/latest/minishift/using/run-against-an-existing-machine.html , mutiple users makes more sense.

**Comment by gbraad** [comment]
Can you summarize what the differences are?
If it is settings based, we might be able to have a seprarate addon to
enable this (and resolve che as a dependency)

**Comment by l0rd** [comment]
@LalatenduMohanty to be honest the main reason we are doing that now is because we want to get rid of the single user Che: it doesn't provide much value and maintaining it requires a lot of resources.

@gbraad the difference is that Che will be bundled with an OpenID Connect provider and a DBMS (that will run in 2 extra containers). Users need to authenticate themselves to use Che. The benefit is that every user will get his own set of protected workspaces (whereas with single user Che anyone that has access to Che URL has access to your workspace, github token etc...). I haven't thought about making it a distinct addon or not yet. Do you suggest to make it a separate one?

**Comment by gbraad** [comment]
It depends on how many changes are needed. I expect running the containers
is not the issue, but likely the needed config change/command might be.
Although I would prefer to see

'che-multi-user' as an add-on to use 'che' as dependency. This is of course
also based on the actual difference. Although this seems minimal.

But if you want to get rid of the single user? On that case, override the
current che and modify this as is. With a note to not to update when single
user is wanted.


-- 

   Gerard Braad | http://gbraad.nl
   [ Doing Open Source Matters ]

**Comment by l0rd** [comment]
This is still relevant

**Comment by l0rd** [comment]
@LalatenduMohanty @gbraad can we re-open this issue?

**Comment by anjannath** [comment]
@l0rd Is this still on the roadmap?

**Comment by l0rd** [comment]
@anjannath yes we still plan to work on it.
